framework/wazuh/core/cluster/tests/test_worker.py

```diff
@@ -120,3 +133,130 @@ def test_remove_bulk_agents(isdir_mock, connection_mock, agents_mock, glob_mock,
+    await check_message(mock=b'False', expected_message="INFO:wazuh:Master didnt grant permission to synchronize\n")
+    await check_message(mock=b'True', expected_message="INFO:wazuh:Worker files sent to master\n")
+
+    error = WazuhException(1001)
+    with patch('wazuh.core.cluster.common.Handler.send_request', new=AsyncMock(return_value=b'True')):
+        with patch('wazuh.core.cluster.common.Handler.send_file', new=AsyncMock(side_effect=error)):
+            await sync_worker.sync()
+            assert b'ERROR:wazuh:Error sending files information' in get_last_logger_line()
+
+    error = KeyError(1)
+    with patch('wazuh.core.cluster.common.Handler.send_request', new=AsyncMock(return_value=b'True')):
+        with patch('wazuh.core.cluster.common.Handler.send_file', new=AsyncMock(side_effect=error)):
+            await sync_worker.sync()
+            assert b'ERROR:wazuh:Error sending files information' in get_last_logger_line()
+
+    os.remove(current_path_logger)
+
+
+def test_WorkerHandler():
+    worker_handler = get_worker_handler()
+    with patch('wazuh.core.cluster.client.AbstractClient.connection_result', side_effect=None):
+        with patch('os.path.exists', return_value=False):
+            with patch('wazuh.core.cluster.worker.utils.mkdir_with_mode', side_effect=None):
+                worker_handler.connected = True
+                worker_handler.connection_result(future_result=None)
+
+    assert worker_handler.process_request(command=b'sync_m_c_ok', data=b'Testing') == (b'ok', b'Thanks')
+
+    with patch('wazuh.core.cluster.common.WazuhCommon.setup_receive_file', side_effect=(b'ok', b'Thanks')):
+        assert worker_handler.process_request(command=b'sync_m_c', data=b'Testing') == b'ok'
+
+    with patch('wazuh.core.cluster.common.WazuhCommon.end_receiving_file',
+               side_effect=(b'ok', b'File correctly received')):
+        assert worker_handler.process_request(command=b'sync_m_c_e', data=b'Testing First') == b'ok'
+
+    with patch('wazuh.core.cluster.common.WazuhCommon.error_receiving_file',
+               side_effect=(b'ok', b'File correctly received')):
+        assert worker_handler.process_request(command=b'sync_m_c_r', data=b'Testing') == b'ok'
+
+    with patch('asyncio.create_task', side_effect=None):
+        assert worker_handler.process_request(command=b'dapi_res', data=b'Testing') == \
+               (b'ok', b'Response forwarded to worker')
+
+        # worker.py:162 local_server is not a member of AbstractClientManager
+        # assert worker_handler.process_request(command=b'dapi_err', data=b'Testing Error') == \
+        #        (b'ok', b'DAPI error forwarded to worker')
+
+        # worker.py:167 dapi is not a member of AbstractClientManager
+        # assert worker_handler.process_request(command=b'dapi', data=b'Testing Error') == \
+        #        (b'ok', b'Added request to API requests queue')

```

Comment:
Should this be commented?

Reply:
This was commented and for the moment we are going to leave it until we define well the behavior of this module.